util.py

```diff
@@ -0,0 +1,12 @@
+def convert_str_to_bool(value: str, default: bool) -> bool:
+    try:
+        if value is None:
+            return default
+        if value.lower() == 'true':
+            return True
+        elif value.lower() == 'false':
+            return False
+        else:
+            return default
+    except Exception as e:
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Broad exception catch</b></div>
<div id="fix">

The 'except' clause is catching a broad 'Exception'. It's generally better to catch specific exceptions that you expect might occur. If you're intentionally catching all exceptions, consider using 'except Exception as e:' and logging the exception for debugging purposes.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -9,5 +9,7 @@
          return False
      else:
          return default
 -    except Exception as e:
 +    except ValueError as e:
 +        # Log the exception for debugging
 +        print(f"Error converting value: {e}")
          return default
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/test/pull/2#issuecomment-2435133415>#342f53</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged